nntrainer/layers/dropout.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file   dropout_layer.cpp
+ * @date   16 June 2020
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug	   No known bugs except for NYI items
+ * @brief  This is Dropout Layer Class for Neural Network
+ *
+ */
+
+#include <dropout.h>
+#include <layer_internal.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int DropOutLayer::initialize(Manager &manager) {
+  output_dim = input_dim;
+  // TODO : asking Tensor to manager
+  for (auto &t : input_dim) {
+    mask.push_back(std::make_shared<Tensor>(t, true));
+  }
+
+  return ML_ERROR_NONE;
+}
+
+void DropOutLayer::forwarding(bool training) {
+  auto &rate_ = std::get<props::DropOutSpec>(dropout_rate).get();
+
+  if (training && rate_ > 0.0) {
```

Comment:
if `rate_ == 0.0`, the output tensors go uninitialized?

Reply:
Current implementation assumes in-place operation. If it is, then it is same with input if rate_== 0.0 or inference mode as well. it means it is doing nothing.